sfs/time/source.py

```diff
@@ -79,13 +79,16 @@ def point(xs, signal, observation_time, grid, c=None):
     if c is None:
         c = default.c
     r = np.linalg.norm(grid - xs)
-    # evaluate g over grid
-    weights = 1 / (4 * np.pi * r)
+    # evaluate g over grid (`r` can get `0`, which is fine)
+    with np.errstate(divide='ignore'):
+        weights = 1 / (4 * np.pi * r)
     delays = r / c
     base_time = observation_time - signal_offset
-    return weights * np.interp(base_time - delays,
-                               np.arange(len(data)) / samplerate,
-                               data, left=0, right=0)
+    # Result can be NaN, which is fine
```

Comment:
What's the actual problem here?
If `weights` is +- inf?

In that case you should probably move the `np.interp()` call out of the `with` statement.

Reply:
`weights` are the output of a possible `1/0` division, which means it can contain `(inf+nanj)`.

Reply:
OK, then the `np.interp()` call should not be in the `with` statement.

Reply:
Changed it to
```python
    points_at_time = np.interp(base_time - delays,
                               np.arange(len(data)) / samplerate,
                               data, left=0, right=0)
    # weights can be +-infinity
    with np.errstate(invalid='ignore'):
        return weights * points_at_time
```